src/biotite/structure/io/gro/file.py

```diff
@@ -171,11 +187,13 @@ def set_structure(self, array):
 
         def get_box_dimen(array):
             """
-            GRO files have the box dimensions as last line for each model.
-            Because we cannot properly detect the box shape, we simply use
-            the min and max coordinates in xyz to get the correct size
+            Box dimensions are mandatory in GRO files. If the array has no
+            associated box dimensions, we set them to 0.
             """
-            return np.abs(array.coord.max(axis=0) - array.coord.min(axis=0))/10
+            if array.box is None:
+                return np.full(9, 0.0)
+            else:
```

Comment:
Small change to the old default case here: The old implementation tried to assume the box dimensions from atom coordinates. However, this is not really helpful and basically just wastes computation time. You either have a box set and this is used, or you have no box (in this case the size is set to 0 in agreement with GROMACS). Assumptions on the box vectors from coordinates cannot account for non-cubic boxes and do not factor in VdW radii so they cant be used in a real world scenario anyways.